lib/SILOptimizer/Utils/Generics.cpp

```diff
@@ -677,7 +677,7 @@ void ReabstractionInfo::createSubstitutedAndSpecializedTypes() {
 
   CanGenericSignature CanSig;
   if (SpecializedGenericSig)
-    CanSig = SpecializedGenericSig->getCanonicalSignature();
+    CanSig = SpecializedGenericSig.getCanonicalSignature();
```

Comment:
And here

Reply:
Thanks for the thorough review!

After removing the conditional, it turns out that `CanSig` is unused:
```
swift/lib/SILOptimizer/Utils/Generics.cpp:678:8: warning: unused variable 'CanSig' [-Wunused-variable]
  auto CanSig = SpecializedGenericSig.getCanonicalSignature();
       ^
```

So I deleted `CanSig` altogether.